### PR TITLE
deps: update repository dependencies and cleanup linters

### PR DESCRIPTION
**Pull request by shoenig**
This PR bulk updates dependencies for the podman driver. It adds a `replace` statement for `nomad`, which includes updates to driver plumbing that we want, which will ship in 1.6. 

It also cleans up the golangci linter file, removing dead linters and adding some fresh ones. Also cleans up the source code in a few places to pass the linter checks. 

Also bumps Go version to 1.20.3

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -2,83 +2,74 @@
...
+    - scripts
 
-  # which files to skip: they will be analyzed, but issues from them
-  # won't be reported. Default value is empty list, but there is
-  # no need to include all autogenerated files, we confidently recognize
-  # autogenerated files. If it's not please let us know.
+  # Skip files
   skip-files:
     - ".*\\.generated\\.go$"
     - ".*bindata_assetfs\\.go$"
```
Comment (lgfa29) [inline .golangci.yml:21]: I think these are copy-pasta from Nomad and could be removed as well. No harm in keeping them thought 🤷

File: driver.go [hunk 2]
```diff
@@ -493,16 +494,17 @@ func (d *Driver) StartTask(cfg *drivers.TaskConfig) (*drivers.TaskHandle, *drive
 	createOpts.ContainerBasicConfig.Labels = driverConfig.Labels
 
 	// Logging
-	if driverConfig.Logging.Driver == "" || driverConfig.Logging.Driver == LOG_DRIVER_NOMAD {
+	switch {
+	case driverConfig.Logging.Driver == "" || driverConfig.Logging.Driver == LOG_DRIVER_NOMAD:
```
Comment (lgfa29) [inline driver.go:498]:
```suggestion
	switch driverConfig.Logging.Driver {
	case "", LOG_DRIVER_NOMAD:
```

Maybe? Seems simpler like this.

File: driver.go [hunk 3]
```diff
@@ -547,9 +549,9 @@ func (d *Driver) StartTask(cfg *drivers.TaskConfig) (*drivers.TaskHandle, *drive
 	}
 
 	if driverConfig.MemorySwap != "" {
-		swap, err := memoryInBytes(driverConfig.MemorySwap)
-		if err != nil {
-			return nil, nil, err
+		swap, memErr := memoryInBytes(driverConfig.MemorySwap)
+		if memErr != nil {
+			return nil, nil, memErr
```
Comment (lgfa29) [inline driver.go:554]: Is using `err` considered variable shadowing in cases like this?

**Review by lgfa29 — APPROVED** [review]
Thanks for the cleanup!

File: driver.go (hunk 2 above)
Comment (shoenig) [reply driver.go:498]: ooh nice catch! i was just bulldozing my way through these >_<

File: driver.go (hunk 3 above)
Comment (shoenig) [reply driver.go:554]: Yup; I'm not sure being this strict about shadowing will turn out for the best, but figured I'd give a try while doing some work on this repo

**Pull request merged**